cvat/apps/engine/utils.py

```diff
@@ -171,9 +176,24 @@ def define_dependent_job(queue: DjangoRQ, user_id: int, should_be_dependent: boo
         for job in queue.job_class.fetch_many(
             queue.deferred_job_registry.get_job_ids(), queue.connection
         )
-        if job and job.meta.get("user", {}).get("id") == user_id
+        if job and job.meta.get("user", {}).get("id") == user_id and job.is_deferred
```

Comment:
We should check each job from `deferred_job_registry` that it is still deferred because `deferred_job_registry.get_job_ids()` calls `deferred_job_registry.cleanup()` which is dummy and has an "empty" implementation

Reply:
I have no exact explanation for that. Need to debug this more, but looks like it can happen if there are active or deferred jobs when restarting the worker container. I have seen several times that one job with `finished` status exists in 2 registries `DeferredJobRegistry` and `FinishedJobRegistry`. FinishedJobRegistry and  StartedJobRegistry have cleanup implementation and expired jobs will be deleted eventually. However, deferred jobs can be deleted from deferred job registry only directly like [here](https://github.com/rq/rq/blob/d11094af95188c197a5e11b1c4fa05b000485370/rq/queue.py#L1224C43-L1224C43). I'll send you screenshots with evidence.

Reply:
Okay. Can you summarize this issue in a code comment?